src/Features/CSharp/Portable/Completion/CompletionProviders/InternalsVisibleToCompletionProvider.cs

```diff
@@ -0,0 +1,145 @@
+            //[Attribute(""|
+            //[Attribute("Text"|)
+            var node = token.Parent;
+            if (node is LiteralExpressionSyntax && node.Kind() == SyntaxKind.StringLiteralExpression)
+            {
+                node = node.Parent;
+                if (node is AttributeArgumentSyntax)
+                {
+                    node = node.Parent;
+                    if (node is AttributeArgumentListSyntax)
+                    {
+                        return node.Parent as AttributeSyntax;
+                    }
+                }
+            }
+            return default(AttributeSyntax);
+        }
+
+        private static void AddAssemblyCompletionItems(CompletionContext context, CancellationToken cancellationToken)
+        {
+            var currentProject = context.Document.Project;
+            foreach (var p in context.Document.Project.Solution.Projects)
+            {
+                if (p == currentProject)
+                {
+                    continue;
+                }
+                var completionItem = CompletionItem.Create(displayText: p.AssemblyName,
+                    properties: ImmutableDictionary.Create<string, string>().Add(ProjectIdKey, p.Id.Id.ToString()));
+                context.AddItem(completionItem);
+            }
+        }
+
+        public override async Task<CompletionChange> GetChangeAsync(Document document, CompletionItem item, char? commitKey = default(char?), CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (item.Properties.TryGetValue(ProjectIdKey, out var projectId))
+            {
+                var assemblyName = item.DisplayText;
+                var project = document.Project.Solution.GetProject(ProjectId.CreateFromSerialized(new System.Guid(projectId)));
+                var publicKey = await GetPublicKeyOfProjectAsync(project, cancellationToken).ConfigureAwait(false);
+                if (!string.IsNullOrEmpty(publicKey))
+                {
+                    assemblyName += ", PublicKey=" + publicKey;
+                }
+                var tc = new TextChange(item.Span, assemblyName);
+                return CompletionChange.Create(tc);
+            }
+            Debug.Fail("Project can't be found by projectId.");
+            return await base.GetChangeAsync(document, item, commitKey, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<string> GetPublicKeyOfProjectAsync(Project p, CancellationToken cancellationToken)
+        {
+            var compilationOptions = p.CompilationOptions;
+            if (compilationOptions == null ||
+                (string.IsNullOrEmpty(compilationOptions.CryptoKeyFile) && string.IsNullOrEmpty(compilationOptions.CryptoKeyContainer)))
+            {
+                return string.Empty;
+            }
+            var c = await p.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
Oh... you need the compilation to compute this?  One way you could make this faster would be to fork the project to one that has no source-files/references.  Then asking for the compilation for it would be trivially cheap.